<!doctype html><html class=no-js lang=en><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><title>RNA Seq 上游分析实践 - yuanj's blog</title><script>(function(e,t){e[t]=e[t].replace("no-js","js")})(document.documentElement,"className")</script><meta name=description content><meta property="og:title" content="RNA Seq 上游分析实践"><meta property="og:description" content="之前那一篇文章主要讲的是一些知识与工具的用法，这次用六组数据进行分析，得到基因表达矩阵。"><meta property="og:type" content="article"><meta property="og:url" content="https://yuanj.top/posts/j9t9v3y1/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2023-09-08T20:29:36+08:00"><meta property="article:modified_time" content="2023-09-08T20:29:36+08:00"><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=dns-prefetch href=//fonts.googleapis.com><link rel=dns-prefetch href=//fonts.gstatic.com><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700"><link rel=stylesheet href=/css/style.css></head><body class=body><div class="container container--outer"><header class=header><script async src=//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js></script><div class="container header__container"><div class=logo><a class=logo__link href=/ title="yuanj's blog" rel=home><div class="logo__item logo__text"><div class=logo__title>yuanj's blog</div><div class=logo__tagline>思绪来得快去得也快，偶尔会在这里停留</div></div></a></div></div></header><div class="wrapper flex"><div class=primary><main class=main role=main><article class=post><header class=post__header><h1 class=post__title>RNA Seq 上游分析实践</h1><div class="post__meta meta"><div class="meta__item-datetime meta__item"><time class=meta__text datetime=2023-09-08T20:29:36+08:00>September 08, 2023</time></div></div><div class=post_busuanzi><p>本文总阅读量 <span id=busuanzi_value_page_pv></span> 次</p></div></header><div class="content post__content clearfix"><p>之前那一篇文章主要讲的是一些知识与工具的用法，这次用六组数据进行分析，得到基因表达矩阵。</p><p>还是先从 NCBI 搜索数据，查看数据是双端测序的数据之后，多选条目，直接 Accession list</p><p><img src=https://jihulab.com/UncleCAT4/static/-/raw/main/blog/2023090820339.png alt></p><p>我使用的是下列数据</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-txt data-lang=txt><span style=display:flex><span>SRR25907783
</span></span><span style=display:flex><span>SRR25907784
</span></span><span style=display:flex><span>SRR25907785
</span></span><span style=display:flex><span>SRR25907786
</span></span><span style=display:flex><span>SRR25907787
</span></span><span style=display:flex><span>SRR25907788
</span></span></code></pre></div><p>使用 sra-tools 批量下载数据，并且使用 nohup 把程序挂在后台下载</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>nohup prefetch --option-file acc_list.txt 
</span></span></code></pre></div><p>再使用脚本进行拆分</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>mkdir SRR
</span></span><span style=display:flex><span>mv ./SRR*/*.sra ./SRR
</span></span><span style=display:flex><span>cd SRR
</span></span><span style=display:flex><span>nohup fastq-dump --gzip --split-3 SRR*.sra 
</span></span></code></pre></div><p>这里千万不要在程序没有运行完成的时候就进行下一步操作，使用 top 可以查看后台进程</p><p><img src=https://jihulab.com/UncleCAT4/static/-/raw/main/blog/20230908204934.png alt></p><p>当程序运行完成后便可看到后台任务中已经没有前面运行的程序了</p><p><img src=https://jihulab.com/UncleCAT4/static/-/raw/main/blog/20230908205137.png alt></p><p>同一目录下的 nohup.out 文件中是后台进程的运行记录</p><p>拆分完成后就需要进行质量检测，使用通配符批量检测，并且将检测报告放在单独一个文件夹以便后面进行压缩</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>mkdir fastqc_report
</span></span><span style=display:flex><span>fastqc SRR2590778*.fastq.gz -o ./fastqc_report
</span></span></code></pre></div><p>完成之后，将检测报告进行压缩，以便下载查看</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>zip -r fastqc_report.zip fastqc_report/*.html
</span></span></code></pre></div><p>依据检测报告对序列进行过滤，参数之前已经讲过，这里数据比较多，写一个 bash 脚本的 for 循环</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span><span style=color:#66d9ef>for</span> i in <span style=color:#f92672>{</span>3..8<span style=color:#f92672>}</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>    trimmomatic PE -threads <span style=color:#ae81ff>1</span> -phred33 SRR2590778<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>_1.fastq.gz SRR2590778<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>_2.fastq.gz -summary oryza_sativa_<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>.summary -baseout SRR2590778<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>.fastq.gz LEADING:3 TRAILING:3 SLIDINGWINDOW:5:20 HEADCROP:15 MINLEN:36
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span></code></pre></div><p>下载水稻的参考基因组和注释文件进行 hisat2 比对</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-57/fasta/oryza_sativa/dna/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz
</span></span><span style=display:flex><span>wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-57/gff3/oryza_sativa/Oryza_sativa.IRGSP-1.0.57.gff3.gz
</span></span></code></pre></div><p>把下载的文件解压后重命名，方便使用</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>gzip -d Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz
</span></span><span style=display:flex><span>mv Oryza_sativa.IRGSP-1.0.dna.toplevel.fa oryza_sativa.fa
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>gzip -d Oryza_sativa.IRGSP-1.0.57.gff3.gz
</span></span><span style=display:flex><span>mv Oryza_sativa.IRGSP-1.0.57.gff3 oryza_sativa.gff3
</span></span></code></pre></div><p>先构建 hisat2 索引</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>hisat2-build oryza_sativa.fa oryza_sativa
</span></span></code></pre></div><p>构建完成之后开始将 reads 比对到参考基因组，还是写成脚本进行</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span><span style=color:#66d9ef>for</span> i in <span style=color:#f92672>{</span>3..8<span style=color:#f92672>}</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>    hisat2 -x hisat2_index/oryza_sativa -p <span style=color:#ae81ff>5</span> -1 SRR2590778<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>_1P.fastq.gz -2 SRR2590778<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>_2P.fastq.gz -S oryza_sativa_<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>.sam
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span></code></pre></div><p>比对完成后要进行排序压缩，到这里时我的 samtools 一直报错，Google 了一下才发现是 samtools 版本的一个 bug，安装指定版本就没问题了，于是新建 conda 环境进行安装（如果没有报错直接使用即可）</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span>conda create -n samtools python<span style=color:#f92672>=</span>3.10
</span></span><span style=display:flex><span>conda search samtools
</span></span><span style=display:flex><span>conda install samtools<span style=color:#f92672>=</span>1.9  <span style=color:#75715e>#指定安装 1.9</span>
</span></span></code></pre></div><p>安装完成继续进行，还是写成 bash 脚本一键进行，这里我将每次排序的结构按照 1-6 分别命名</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span><span style=color:#66d9ef>for</span> i in <span style=color:#f92672>{</span>3..8<span style=color:#f92672>}</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>    samtools sort -n -@ <span style=color:#ae81ff>5</span> oryza_sativa_<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>.sam -o oryza_sativa_<span style=color:#e6db74>${</span>i<span style=color:#e6db74>}</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span></code></pre></div><p>最后一步 featureCounts 生成基因计数表</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-bash data-lang=bash><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>bam_files<span style=color:#f92672>=(</span>*.bam<span style=color:#f92672>)</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>if</span> <span style=color:#f92672>[</span> <span style=color:#e6db74>${#</span>bam_files[@]<span style=color:#e6db74>}</span> -gt <span style=color:#ae81ff>0</span> <span style=color:#f92672>]</span>; <span style=color:#66d9ef>then</span>
</span></span><span style=display:flex><span>    featureCounts -T <span style=color:#ae81ff>5</span> -t exon -g Name -a Lolium_perenne.gff3 -o  gene.counts -p <span style=color:#e6db74>&#34;</span><span style=color:#e6db74>${</span>bam_files[@]<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>else</span>
</span></span><span style=display:flex><span>    echo <span style=color:#e6db74>&#34;No BAM files found in the current directory.&#34;</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>fi</span>
</span></span></code></pre></div><p>至此完成，拿到矩阵</p><p>个人习惯完成分析后把文件进行一下归类</p><p><img src=https://jihulab.com/UncleCAT4/static/-/raw/main/blog/20230908205327.png alt></p><p>最后将数据下载到本地进行下游分析</p><p><img src=https://jihulab.com/UncleCAT4/static/-/raw/main/blog/2023090820577.png alt></p></div></article></main><script src=https://giscus.app/client.js data-repo=UncleCAT4/blog data-repo-id=R_kgDOKXKUsQ data-category=Announcements data-category-id=DIC_kwDOKXKUsc4CZj38 data-mapping=title data-strict=0 data-reactions-enabled=0 data-emit-metadata=0 data-input-position=bottom data-theme=preferred_color_scheme data-lang=en crossorigin=anonymous async></script></div></div></div><script async defer src=/js/menu.js></script></body></html>