<!doctype html><html lang=en-US><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="chrome=1"><meta name=HandheldFriendly content="True"><meta name=MobileOptimized content="320"><meta name=viewport content="width=device-width,initial-scale=1"><meta name=referrer content="no-referrer"><meta name=author content="YuanJ2003"><meta name=description content="之前那一篇文章主要讲的是一些知识与工具的用法，这次用六组数据进行分析，得到基因表达矩阵
还是先从 NCBI 搜索数据，查看数据是双端测序的数据之后，多选条目，直接 Accession list
我使用的是下列数据
SRR25907783 SRR25907784 SRR25907785 SRR25907786 SRR25907787 SRR25907788 使用 sra-tools 批量下载数据，并且使用 nohup 把程序挂在后台下载
nohup prefetch --option-file acc_list.txt 再使用脚本进行拆分
#!/bin/bash mkdir SRR mv ./SRR*/*.sra ./SRR cd SRR nohup fastq-dump --gzip --split-3 SRR*.sra 这里千万不要在程序没有运行完成的时候就进行下一步操作，使用 top 可以查看后台进程
当程序运行完成后便可看到后台任务中已经没有前面运行的程序了
同一目录下的 nohup.out 文件中是后台进程的运行记录
拆分完成后就需要进行质量检测，使用通配符批量检测，并且将检测报告放在单独一个文件夹以便后面进行压缩
mkdir fastqc_report fastqc SRR2590778*.fastq.gz -o ./fastqc_report 完成之后，将检测报告进行压缩，以便下载查看
zip -r fastqc_report.zip fastqc_report/*.html 依据检测报告对序列进行过滤，参数之前已经讲过，这里数据比较多，写一个 bash 脚本的 for 循环
#!/bin/bash for i in {3..8} do trimmomatic PE -threads 1 -phred33 SRR2590778${i}_1."><meta property="og:title" content="RNA Seq 上游分析实践"><meta property="og:description" content="之前那一篇文章主要讲的是一些知识与工具的用法，这次用六组数据进行分析，得到基因表达矩阵
还是先从 NCBI 搜索数据，查看数据是双端测序的数据之后，多选条目，直接 Accession list
我使用的是下列数据
SRR25907783 SRR25907784 SRR25907785 SRR25907786 SRR25907787 SRR25907788 使用 sra-tools 批量下载数据，并且使用 nohup 把程序挂在后台下载
nohup prefetch --option-file acc_list.txt 再使用脚本进行拆分
#!/bin/bash mkdir SRR mv ./SRR*/*.sra ./SRR cd SRR nohup fastq-dump --gzip --split-3 SRR*.sra 这里千万不要在程序没有运行完成的时候就进行下一步操作，使用 top 可以查看后台进程
当程序运行完成后便可看到后台任务中已经没有前面运行的程序了
同一目录下的 nohup.out 文件中是后台进程的运行记录
拆分完成后就需要进行质量检测，使用通配符批量检测，并且将检测报告放在单独一个文件夹以便后面进行压缩
mkdir fastqc_report fastqc SRR2590778*.fastq.gz -o ./fastqc_report 完成之后，将检测报告进行压缩，以便下载查看
zip -r fastqc_report.zip fastqc_report/*.html 依据检测报告对序列进行过滤，参数之前已经讲过，这里数据比较多，写一个 bash 脚本的 for 循环
#!/bin/bash for i in {3..8} do trimmomatic PE -threads 1 -phred33 SRR2590778${i}_1."><meta property="og:type" content="article"><meta property="og:url" content="https://hieroglyphs.top/posts/kjasdf98/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2023-09-08T20:29:36+08:00"><meta property="article:modified_time" content="2023-09-08T20:29:36+08:00"><title>RNA Seq 上游分析实践</title><link rel=canonical href=https://hieroglyphs.top/posts/kjasdf98/><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.11.0/styles/default.min.css><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Open+Sans:400,700|Ubuntu+Mono:400,400i,700,700i|Raleway:500"><link rel=stylesheet href="https://hieroglyphs.top/css/reset.css?t=2023-10-05%2002%3a33%3a17.421866789%20%2b0000%20UTC%20m%3d%2b0.111444648"><link rel=stylesheet href="https://hieroglyphs.top/css/pygments.css?t=2023-10-05%2002%3a33%3a17.421866789%20%2b0000%20UTC%20m%3d%2b0.111444648"><link rel=stylesheet href="https://hieroglyphs.top/css/main.css?t=2023-10-05%2002%3a33%3a17.421866789%20%2b0000%20UTC%20m%3d%2b0.111444648"><link rel=stylesheet href="?t=2023-10-05%2002%3a33%3a17.421866789%20%2b0000%20UTC%20m%3d%2b0.111444648"><link rel="shortcut icon" href=https://hieroglyphs.top/images/favicon.ico></head><body lang=en><section class=header><div class=container><div class=content><a href=https://hieroglyphs.top/><div class=name>YuanJ2003</div></a><h3 class=self-intro>Bioinformatics | HJLab @ CDUT | Chengdu</h3><nav><ul><li class=nav-posts><a href=https://hieroglyphs.top/posts/><span>Posts</span></a></li><li class=nav-diary><a href=https://hieroglyphs.top/diary/><span>Diary</span></a></li><li class=nav-about><a href=https://hieroglyphs.top/about/><span>About</span></a></li><li class=nav-gestbook><a href=https://hieroglyphs.top/gestbook/><span>Gestbook</span></a></li><li class=nav-links><a href=https://hieroglyphs.top/links/><span>Links</span></a></li></ul></nav></div></div></section><section class=icons><div class=container><div class=content><a href=//github.com/YuanJ2003 target=_blank rel=noopener><img class=icon src=https://hieroglyphs.top/icons/github.svg alt=github></a>
<a href=https://hieroglyphs.top/images/email.png target=_blank rel=noopener><img class=icon src=https://hieroglyphs.top/icons/email.svg alt=email></a>
<a href=https://hieroglyphs.top/images/wechat.jpg><img class=icon src=https://hieroglyphs.top/icons/wechat.svg alt=wechat></a>
<a href="//space.bilibili.com/1692277850?spm_id_from=333.1007.0.0"><img class=icon src=https://hieroglyphs.top/icons/bilibili.svg alt=bilibili></a>
<a href=https://hieroglyphs.top/index.xml><img class=icon src=https://hieroglyphs.top/icons/rss.svg alt=rss></a></div></div></section><section class="main post non-narrow zero-top-spacing"><div class=container><div class=content><div class=front-matter><div class=title-container><div class=page-heading>RNA Seq 上游分析实践</div><div class=initials><a href=https://hieroglyphs.top>ad</a></div></div><div class=meta><div class=date title='Fri Sep 8 2023 20:29:36 +0800'>Sep 8, 2023</div></div></div><div class=markdown><blockquote><p>之前那一篇文章主要讲的是一些知识与工具的用法，这次用六组数据进行分析，得到基因表达矩阵</p></blockquote><p>还是先从 NCBI 搜索数据，查看数据是双端测序的数据之后，多选条目，直接 Accession list</p><p><img src=https://jihulab.com/YuanJ2003/static/-/raw/main/blog-images/2023090820339.png alt></p><p>我使用的是下列数据</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-txt data-lang=txt><span class=line><span class=cl>SRR25907783
</span></span><span class=line><span class=cl>SRR25907784
</span></span><span class=line><span class=cl>SRR25907785
</span></span><span class=line><span class=cl>SRR25907786
</span></span><span class=line><span class=cl>SRR25907787
</span></span><span class=line><span class=cl>SRR25907788
</span></span></code></pre></div><p>使用 sra-tools 批量下载数据，并且使用 nohup 把程序挂在后台下载</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>nohup prefetch --option-file acc_list.txt 
</span></span></code></pre></div><p>再使用脚本进行拆分</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl><span class=cp>#!/bin/bash
</span></span></span><span class=line><span class=cl><span class=cp></span>mkdir SRR
</span></span><span class=line><span class=cl>mv ./SRR*/*.sra ./SRR
</span></span><span class=line><span class=cl><span class=nb>cd</span> SRR
</span></span><span class=line><span class=cl>nohup fastq-dump --gzip --split-3 SRR*.sra 
</span></span></code></pre></div><p>这里千万不要在程序没有运行完成的时候就进行下一步操作，使用 top 可以查看后台进程</p><p><img src=https://jihulab.com/YuanJ2003/static/-/raw/main/blog-images/20230908204934.png alt></p><p>当程序运行完成后便可看到后台任务中已经没有前面运行的程序了</p><p><img src=https://jihulab.com/YuanJ2003/static/-/raw/main/blog-images/20230908205137.png alt></p><p>同一目录下的 nohup.out 文件中是后台进程的运行记录</p><p>拆分完成后就需要进行质量检测，使用通配符批量检测，并且将检测报告放在单独一个文件夹以便后面进行压缩</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>mkdir fastqc_report
</span></span><span class=line><span class=cl>fastqc SRR2590778*.fastq.gz -o ./fastqc_report
</span></span></code></pre></div><p>完成之后，将检测报告进行压缩，以便下载查看</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>zip -r fastqc_report.zip fastqc_report/*.html
</span></span></code></pre></div><p>依据检测报告对序列进行过滤，参数之前已经讲过，这里数据比较多，写一个 bash 脚本的 for 循环</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl><span class=cp>#!/bin/bash
</span></span></span><span class=line><span class=cl><span class=cp></span><span class=k>for</span> i in <span class=o>{</span>3..8<span class=o>}</span>
</span></span><span class=line><span class=cl><span class=k>do</span>
</span></span><span class=line><span class=cl>    trimmomatic PE -threads <span class=m>1</span> -phred33 SRR2590778<span class=si>${</span><span class=nv>i</span><span class=si>}</span>_1.fastq.gz SRR2590778<span class=si>${</span><span class=nv>i</span><span class=si>}</span>_2.fastq.gz -summary oryza_sativa_<span class=si>${</span><span class=nv>i</span><span class=si>}</span>.summary -baseout SRR2590778<span class=si>${</span><span class=nv>i</span><span class=si>}</span>.fastq.gz LEADING:3 TRAILING:3 SLIDINGWINDOW:5:20 HEADCROP:15 MINLEN:36
</span></span><span class=line><span class=cl><span class=k>done</span>
</span></span></code></pre></div><p>下载水稻的参考基因组和注释文件进行 hisat2 比对</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-57/fasta/oryza_sativa/dna/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz
</span></span><span class=line><span class=cl>wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-57/gff3/oryza_sativa/Oryza_sativa.IRGSP-1.0.57.gff3.gz
</span></span></code></pre></div><p>把下载的文件解压后重命名，方便使用</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>gzip -d Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz
</span></span><span class=line><span class=cl>mv Oryza_sativa.IRGSP-1.0.dna.toplevel.fa oryza_sativa.fa
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl>gzip -d Oryza_sativa.IRGSP-1.0.57.gff3.gz
</span></span><span class=line><span class=cl>mv Oryza_sativa.IRGSP-1.0.57.gff3 oryza_sativa.gff3
</span></span></code></pre></div><p>先构建 hisat2 索引</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>hisat2-build oryza_sativa.fa oryza_sativa
</span></span></code></pre></div><p>构建完成之后开始将 reads 比对到参考基因组，还是写成脚本进行</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl><span class=cp>#!/bin/bash
</span></span></span><span class=line><span class=cl><span class=cp></span><span class=k>for</span> i in <span class=o>{</span>3..8<span class=o>}</span>
</span></span><span class=line><span class=cl><span class=k>do</span>
</span></span><span class=line><span class=cl>    hisat2 -x hisat2_index/oryza_sativa -p <span class=m>5</span> -1 SRR2590778<span class=si>${</span><span class=nv>i</span><span class=si>}</span>_1P.fastq.gz -2 SRR2590778<span class=si>${</span><span class=nv>i</span><span class=si>}</span>_2P.fastq.gz -S oryza_sativa_<span class=si>${</span><span class=nv>i</span><span class=si>}</span>.sam
</span></span><span class=line><span class=cl><span class=k>done</span>
</span></span></code></pre></div><p>比对完成后要进行排序压缩，到这里时我的 samtools 一直报错，Google 了一下才发现是 samtools 版本的一个 bug，安装指定版本就没问题了，于是新建 conda 环境进行安装</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl>conda create -n samtools <span class=nv>python</span><span class=o>=</span>3.10
</span></span><span class=line><span class=cl>conda search samtools
</span></span><span class=line><span class=cl>conda install <span class=nv>samtools</span><span class=o>=</span>1.9  <span class=c1>#指定安装 1.9</span>
</span></span></code></pre></div><p>安装完成继续进行，还是写成 bash 脚本一键进行，这里我将每次排序的结构按照 1-6 分别命名</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl><span class=cp>#!/bin/bash
</span></span></span><span class=line><span class=cl><span class=cp></span><span class=k>for</span> i in <span class=o>{</span>3..8<span class=o>}</span>
</span></span><span class=line><span class=cl><span class=k>do</span>
</span></span><span class=line><span class=cl>    samtools sort -n -@ <span class=m>5</span> oryza_sativa_<span class=si>${</span><span class=nv>i</span><span class=si>}</span>.sam -o oryza_sativa_<span class=si>${</span><span class=nv>i</span><span class=si>}</span>
</span></span><span class=line><span class=cl><span class=k>done</span>
</span></span></code></pre></div><p>最后一步 featureCounts 生成基因计数表</p><div class=highlight><pre tabindex=0 class=chroma><code class=language-bash data-lang=bash><span class=line><span class=cl><span class=cp>#!/bin/bash
</span></span></span><span class=line><span class=cl><span class=cp></span><span class=k>for</span> i in <span class=o>{</span>3..8<span class=o>}</span>
</span></span><span class=line><span class=cl><span class=k>do</span>
</span></span><span class=line><span class=cl>    featureCounts -T <span class=m>5</span> -t exon -g Name -a oryza_sativa.gff3 -o gene.counts -p oryza_sativa_<span class=si>${</span><span class=nv>i</span><span class=si>}</span>
</span></span><span class=line><span class=cl><span class=k>done</span>
</span></span></code></pre></div><p>至此完成，拿到矩阵</p><p>个人习惯完成分析后把文件进行一下归类</p><p><img src=https://jihulab.com/YuanJ2003/static/-/raw/main/blog-images/20230908205327.png alt></p><p>最后将数据下载到本地进行下游分析</p><p><img src=https://jihulab.com/YuanJ2003/static/-/raw/main/blog-images/2023090820577.png alt></p><br><p class=back-to-posts><a href=https://hieroglyphs.top/posts>Back to posts</a></p></div><br></div></div></section><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.11.0/highlight.min.js></script>
<script type=text/javascript>hljs.initHighlightingOnLoad()</script><script type=text/javascript async src="https://cdn.bootcss.com/mathjax/2.7.3/MathJax.js?config=TeX-AMS-MML_HTMLorMML">MathJax.Hub.Config({tex2jax:{inlineMath:[["$","$"],["\\(","\\)"]],displayMath:[["$$","$$"],["[[","]]"]],processEscapes:!0,processEnvironments:!0,skipTags:["script","noscript","style","textarea","pre"],TeX:{equationNumbers:{autoNumber:"AMS"},extensions:["AMSmath.js","AMSsymbols.js"]}}}),MathJax.Hub.Queue(function(){var e,t=MathJax.Hub.getAllJax();for(e=0;e<t.length;e+=1)t[e].SourceElement().parentNode.className+=" has-jax"})</script><style>code.has-jax{font:inherit;font-size:100%;background:inherit;border:inherit;color:#515151}</style></body></html>